### DISCUSSION TITLE
Self-hosted Docker + OAuth Setup

### DISCUSSION THREAD

**Discussion by marcusguttenplan** [supabase/supabase#1311] [Questions] [answered]
Amazing platform and product, and very much enjoying using this for a real-time asset tracker!! 

Currently have integrated super seamlessly with Rails and a Nuxt app using basic pw auth, but would like to extend this to include OAuth providers.

Are the env vars to set for the various providers' secrets? Would these go into the Kong config, or is there another way?

Eventually we'll be moving to the paid hosted platform, but self-hosting is a great way to get acquainted with the code and in the future migrate to k8s or something (data persistence is the only concern there right now).

**Comment by easy-street** [comment, marked as answer]
the OAuth env vars can be added to `docker-compose.yml` under `services.auth.environment`

some relevant resources:
- auth provider configuration docs: https://github.com/supabase/gotrue#external-authentication-providers
- self-hosting service configuration docs: https://supabase.io/docs/guides/self-hosting#configuring-each-service
